lib/rucio/daemons/rsedecommissioner/profiles/generic.py

```diff
@@ -0,0 +1,477 @@
+
+
+def generic_move(
+    rse: 'dict[str, Any]',
+    config: 'dict[str, Any]',
+    *,
+    logger: 'Callable[..., None]' = logging.log
+) -> None:
+    """Move rules that satisfy conditions to a specific destination.
+
+    The "generic move" profile lists out all rules that are locking replicas at the given RSE,
+    and moves them to the specified destination if either one of the following is true:
+
+    - The RSE expression of the rule is trivial (the RSE name itself).
+    - There are no replicas locked by the rule that reside on another RSE.
+
+    :param rse: RSE table entry as a dictionary.
+    :param config: Decommissioning configuration dictionary.
+    :param logger: Logging function.
+    """
+    move_to_destination = RuleMover(config['destination'])
+    handlers = [
+        (_is_being_deleted, _count_as_processed),
+        (_has_child_rule_id, _count_as_processed),
+        (_has_trivial_rse_expression, move_to_destination),
+        (_has_last_replica_on_rse, move_to_destination),
+    ]
+    generic_profile_template_function(rse, config,
+                                      _generic_initialize,
+                                      list_rules_for_rse_decommissioning,
+                                      handlers,
+                                      _generic_finalize,
+                                      logger=logger)
+
+
+def _generic_initialize(
+    rse: 'dict[str, Any]',
+    status: 'DecommissioningStatus',
+    *,
+    logger: 'Callable[..., None]' = logging.log,
+) -> None:
+    """Set the RSE availability flags and the decommissioning status.
+
+    When an RSE is initialized for decommissioning, the availability flags are set as follows:
+
+    - `availability_read=True`
+    - `availability_write=False`
+    - `availability_delete=True`
+
+    and the following attribute is set:
+
+    - `greedyDeletion=True`
+
+    :param rse: RSE table entry as a dictionary.
+    :param status: Decommissioning status.
+    :param logger: Logging function.
+    """
+
+    if status != DecommissioningStatus.PROCESSING:
+        return
+
+    # Set the RSE availability and greedyDeletion attributes.
+    logger(logging.INFO,
+           '(%s) Setting RSE availability flags of to read !write delete, greedyDeletion=true.',
+           rse['rse'])
+
+    parameters = {
+        'availability_read': True,
+        'availability_write': False,
+        'availability_delete': True
+    }
+    update_rse(rse['id'], parameters)
+
+    try:
+        add_rse_attribute(rse['id'], 'greedyDeletion', True)
+    except Duplicate:
+        pass
+
+
+def _generic_finalize(
+    rse: 'dict[str, Any]',
+    *,
+    logger: 'Callable[..., None]' = logging.log
+) -> None:
+    """Check for remaining replicas at the RSE and resolve where possible.
+
+    :param rse: RSE table entry as a dictionary.
+    :param logger: Logging function.
+    :returns: A boolean flag indicating the completion status.
+    """
+    initial_limit = 100
```

Comment:
Maybe this will be better as a keyword-only parameter?

Reply:
Renamed to `list_max` (I don't remember why it was named as initial_limit; there is nothing "initial" about the parameter) and made it into a keyword-only parameter.